docs/userGuide/addingNavigationButtons.md

```diff
@@ -0,0 +1,157 @@
+<div class="indented">
+
+{% raw %}
+```html
+{% macro previous_next(previous_page, next_page) %}
+<div class="clearfix">
+{% if previous_page != ''%}
+<span class="float-start">
+    <a class="btn btn-light" href="{{ previous_page }}.html">
+        <md>:far-arrow-alt-circle-left: <include src="{{ previous_page }}.md#title" inline />
+        </md>
+    </a>
+</span>
+{% endif %}
+{% if next_page != ''%}
+<span class="float-end">
+    <a class="btn btn-light" href="{{ next_page }}.html">
+        <md>
+            <include src="{{ next_page }}.md#title" inline /> :far-arrow-alt-circle-right:
+        </md>
+    </a>
+</span>
+{% endif %}
+</div>
+<br>
+{% endmacro %}
+```
+{% endraw %}
+</div>
+
+{{ icon_example }} Here is an example of how the default template's directory structure will look like after the changes.
+
+<div class="indented">
+
+```
+├── _markbind
+├── _site
+├── contents
+├── njk
+│   ├── common.njk
+├── stylesheets
+├── index.md
+├── site.json
+```
+</div>
+
+## Editing the content files
+
+1. Add the following code into the `topic1.md` file:
+<div class="indented">
+
+{% raw %}
+```html {heading="**topic1.md**"}
+<!--- Add this to the top of the file -->
+<span id="title" class="d-none">Topic 1</span>
+
+<!--- Add this to the end of the file -->
+{% from "njk/common.njk" import previous_next %}
+{{ previous_next('', 'topic2') }}
+```
+{% endraw %}
+</div>
+
+2. Add the following code into the `topic2.md` file:
+<div class="indented">
+
+{% raw %}
+```html {heading="**topic2.md**"}
+<!--- Add this to the top of the file -->
+<span id="title" class="d-none">Topic 2</span>
+
+<!--- Add this to the end of the file -->
+{% from "njk/common.njk" import previous_next %}
+{{ previous_next('topic1', '') }}
+```
+{% endraw %}
+</div>
+
+{{ icon_example }} Here is an example of how the files will look like after the changes.
+
+<div class="indented">
+
+{% raw %}
+```html {heading="**topic1.md**" .line-numbers}
+<span id="title" class="d-none">Topic 1</span>
+
+<br>
+
+# Topic 1
+
+> More content to be added
+
+{% from "njk/common.njk" import previous_next %}
+{{ previous_next('', 'topic2') }}
+```
+{% endraw %}
+
+{% raw %}
+```html {heading="**topic2.md**" .line-numbers}
+<span id="title" class="d-none">Topic 2</span>
+
+<br>
+<box>
+    <span class="fas fa-tools"></span><span> This is a placeholder page</span>
+</box>
+
+{% from "njk/common.njk" import previous_next %}
+{{ previous_next('topic1', '') }}
+```
+{% endraw %}
+
+</div>
+
+You should now be able to navigate between section `Topic 1` and `Topic 2` with ease.
+
+<box type="tip">
+
+Whenever you create a new page, be sure to include the following code below and replace the parts enclosed in the square brackets e.g `[Name_Of_Section]`   
+
+{% raw %}
+```html
+<span id="title" class="d-none">[Name_Of_Section]</span>
+
+{% from "njk/common.njk" import previous_next %}
+{{ previous_next('[Previous_Page_Filename]', '[Next_Page_Filename]') }}
+```
+{% endraw %}
+
+</box>
```

Comment:
Could we move expand on this and move this up before the examples to give readers an understanding of what the code is doing? Mentioning the parts that readers need to change as well as which parts of the UI are affected will give them a better idea of what to do when customizing for their own website.

Also because this whole feature involves njk syntax that readers may not be aware of, we could link some relevant resources, and be a bit more explicit when mentioning what we are doing in the code, like defining an invisible title fragment that's used in the common.njk (same goes for the other important concepts)

Reply:
Oh yes, I will work on that but I have a question regarding how I should go about explaining the idea behind what the code does. I thought of writing it down as a markdown comment above the different parts that require explanation, not sure if you think that's advisable? E.g `defining an invisible title fragment that's used in the common.njk` would be placed above the `<span id="title" class="d-none">[Name_Of_Section]</span>` in the form of a comment

Reply:
Hi @xSaints19x, thank you for moving the tip up. I think it is clearer now but would be great if we can improve it further. Here are my suggestions in more greater detail, please let me know if it works for you:
1. Adding an overview of the whole process here before listing out the steps. What we can add here is a brief introduction as to how the entire thing works. So perhaps along the lines of "we will be making use of nunjucks templates (link to https://markbind.org/userGuide/markBindSyntaxOverview.html#support-for-nunjucks) .... and it works by defining xxx...)
![image](https://user-images.githubusercontent.com/41845017/208028671-fd7b95ff-8d15-440b-bec0-2813d32f9ed7.png)
2. Refactor the paragraph
![image](https://user-images.githubusercontent.com/41845017/208029351-8c58d08c-4f41-4ab9-aac8-9230dcb56986.png)
- it should not be a tip or closable box and the information is quite essential to understanding the code snippet
- avoid using `refer to explanation point 3 below -->`, if an author copies this code snippet, he/she has to painstakely remove the comments that are not appropriate when appearing in their code file. Would be great if the comments are self-sufficient.
- minor edits on capitalization "Effects after changes ..." and not "Effects after Changes..."
- make use of the line number feature and address the line number in your explanation instead of the more verbose "that line of code ...."

Reply:
Sounds good 👍 Let's try that